### ISSUE TITLE
pysam does not install on windows

### ISSUE THREAD

**Issue by AndreasHeger** [pysam-developers/pysam#575]
Pysam does not yet support windows, though there are seem to be mods in the wild that do.

This ticket collects all other tickets until Windows installation has been resolved.

**Comment by bmatern** [comment]
This is still a major obstacle for me. I don't think I understand the scope of the problem, or if I am able to contribute, is there a good summary of the problems? Why doesn't it work?

**Comment by betteridiot** [comment]
This is not meant to sound like advertising, but more of a workaround:

 I recently created a Pure Python BAM file parser and random access tool that is based heavily on the pysam API. Since it doesn't use htslib, it works on all platforms. This package can be found on conda, pypi, or github under the name [`bamnostic`](https://github.com/betteridiot/bamnostic).

It was built specifically to scratch the Windows itch that htslib precludes. It doesn't support CRAM, VCF, or plain text SAM, or CSI indexing yet, but the basic BAM functionality is there.

**Comment by AndreasHeger** [comment]
Thanks for letting everyone know, good to hear than bamnostic is there.

**Comment by Ahhgust** [comment]
For what it's worth, pysam installs just fine on Windows 10 if you use Windows Subsystem for Linux (WSL).

Follow the instructions here to install WSL:

https://docs.microsoft.com/en-us/windows/wsl/install-win10

and choose Ubuntu 18.04LTS

Then run the Ubuntu "app", and run the following:
sudo apt update && upgrade
sudo apt upgrade
sudo apt install python3 python3-pip ipython3
sudo apt install python3-numpy
sudo apt install tabix libhts-dev libseqlib-dev
sudo apt install zlibc
sudo pip3 install pysam

I don't guarantee that the above is minimal, but pysam appears to be working just fine! It's a good bet that this approach works with other linux distributions through WSL.

-August

**Comment by betteridiot** [comment]
This is true, but not all users can use WSL. Again, bamnostic is only mean to diversify the accessibility of users.

**Comment by zofile** [comment]
> For what it's worth, pysam installs just fine on Windows 10 if you use Windows Subsystem for Linux (WSL).
> 
> Follow the instructions here to install WSL:
> 
> https://docs.microsoft.com/en-us/windows/wsl/install-win10
> 
> and choose Ubuntu 18.04LTS
> 
> Then run the Ubuntu "app", and run the following:
> sudo apt update && upgrade
> sudo apt upgrade
> sudo apt install python3 python3-pip ipython3
> sudo apt install python3-numpy
> sudo apt install tabix libhts-dev libseqlib-dev
> sudo apt install zlibc
> sudo pip3 install pysam
> 
> I don't guarantee that the above is minimal, but pysam appears to be working just fine! It's a good bet that this approach works with other linux distributions through WSL.
> 
> -August

It'd been a long time, however I wanted to thank you so much, this was very very helpful for me!

**Comment by kconkas** [comment]
> For what it's worth, pysam installs just fine on Windows 10 if you use Windows Subsystem for Linux (WSL).

This is, literally, installation on Linux. What if someone needs a Windows native module?

**Comment by Boehlern** [comment]
> For what it's worth, pysam installs just fine on Windows 10 if you use Windows Subsystem for Linux (WSL).
> 
> Follow the instructions here to install WSL:
> 
> https://docs.microsoft.com/en-us/windows/wsl/install-win10
> 
> and choose Ubuntu 18.04LTS
> 
> Then run the Ubuntu "app", and run the following: sudo apt update && upgrade sudo apt upgrade sudo apt install python3 python3-pip ipython3 sudo apt install python3-numpy sudo apt install tabix libhts-dev libseqlib-dev sudo apt install zlibc sudo pip3 install pysam
> 
> I don't guarantee that the above is minimal, but pysam appears to be working just fine! It's a good bet that this approach works with other linux distributions through WSL.
> 
> -August

> I would like to second zofile's thanks for this post. This helped me as well. Notably, I was unable to get Pysam to install using WSL until I also installed the libbz2-dev package. But afterwards it installed no problem!

**Comment by kleagsand** [comment]
@Boehlern the tip on libbz2-dev helped enormously as well, thank you!!

**Comment by dj94364** [comment]
> > For what it's worth, pysam installs just fine on Windows 10 if you use Windows Subsystem for Linux (WSL).
> > Follow the instructions here to install WSL:
> > https://docs.microsoft.com/en-us/windows/wsl/install-win10
> > and choose Ubuntu 18.04LTS
> > Then run the Ubuntu "app", and run the following: sudo apt update && upgrade sudo apt upgrade sudo apt install python3 python3-pip ipython3 sudo apt install python3-numpy sudo apt install tabix libhts-dev libseqlib-dev sudo apt install zlibc sudo pip3 install pysam
> > I don't guarantee that the above is minimal, but pysam appears to be working just fine! It's a good bet that this approach works with other linux distributions through WSL.
> > -August
> 
> I would like to second zofile's thanks for this post. This helped me as well. Notably, I was unable to get Pysam to install using WSL until I also installed the libbz2-dev package. But afterwards it installed no problem!

I have installed pysam along with the libbz2-dev package on my WSL/Ubuntu, but I still can't get a package I need in anaconda to successfully install since it is running into issues with pysam. My output from installing pysam on Ubuntu looks like this

$ sudo pip3 install pysam
Collecting pysam
  Using cached pysam-0.19.1.tar.gz (3.9 MB)
Building wheels for collected packages: pysam
  Building wheel for pysam (setup.py) ... done
  Created wheel for pysam: filename=pysam-0.19.1-cp38-cp38-linux_x86_64.whl size=13875483 sha256=9178bf9396b7e5740dd7671f31f17012e52cc400cd8c90ffb91cfb72f3a05d8f
  Stored in directory: /root/.cache/pip/wheels/78/d9/98/da2c7de484c0c4992a625d27d8a81ddb7ca48c0ff5018818d2
Successfully built pysam
Installing collected packages: pysam
Successfully installed pysam-0.19.1

Do you have any suggestions for how I can get my package in anaconda to successfully install/use pysam?
